webapp/src/app.tsx

```diff
@@ -1068,19 +1068,40 @@ export class ProjectView
             .done();
     }
 
-    showDeviceNotFoundDialogAsync(docPath: string): Promise<void> {
+    showDeviceNotFoundDialogAsync(docPath: string, resp: pxtc.CompileResult): Promise<void> {
         pxt.tickEvent(`compile.devicenotfound`);
+
+        let fileType = ".mkcd";
+
+        if (pxt.appTarget.compile.useUF2) {
+            fileType = ".uf2";
+        } else if (pxt.appTarget.compile.hasHex) {
+            fileType = ".hex";
+        }
+
         return core.dialogAsync({
             header: lf("Oops, we couldn't find your {0}", pxt.appTarget.appTheme.boardName),
             body: lf("Please make sure your {0} is connected and try again.", pxt.appTarget.appTheme.boardName),
-            buttons: [{
-                label: lf("Troubleshoot"),
-                class: "focused",
-                url: docPath,
-                onclick: () => {
-                    pxt.tickEvent(`compile.troubleshoot`);
+            buttons: [
+                {
+                    label: lf("Troubleshoot"),
+                    class: "focused",
+                    icon: "help",
+                    url: docPath,
+                    onclick: () => {
+                        pxt.tickEvent(`compile.troubleshoot`);
+                    }
+                },
+                {
+                    label: lf("Download {0} file", fileType),
```

Comment:
Just put the file name

Reply:
`Download [filename]`?

Reply:
Only filename to be consistent with https://github.com/Microsoft/pxt/blob/321b7adfb7c36ac9cc60cf7d4e3d6d7e092544c2/webapp/src/cmds.ts#L92